presto-tests/src/main/java/com/facebook/presto/tests/DistributedQueryRunner.java

```diff
@@ -413,6 +417,39 @@ private NodeState getCoordinatorInfoState(int coordinator)
         return state;
     }
 
+    public NodeState getWorkerInfoState(int worker)
+    {
+        URI uri = URI.create(getWorker(worker).getBaseUrl().toString() + "/v1/info/state");
+        Request request = prepareGet()
+                .setHeader(PRESTO_USER, DEFAULT_USER)
+                .setUri(uri)
+                .build();
+
+        NodeState state = client.execute(request, createJsonResponseHandler(jsonCodec(NodeState.class)));
+        return state;
+    }
+
+    public int sendWorkerShutdown(int worker, String body)
```

Comment:
nit: It is odd to allow a body for the Shutdown.
May be this should be renamed as a generic `sendWorkerRequest`